pkg/controller/machineset/controller.go

```diff
@@ -99,45 +123,60 @@ func (c *MachineSetControllerImpl) Get(namespace, name string) (*v1alpha1.Machin
 }
 
 // syncReplicas essentially scales machine resources up and down.
-func (c *MachineSetControllerImpl) syncReplicas(machineSet *v1alpha1.MachineSet, machines []*v1alpha1.Machine) error {
-	// Take ownership of machines if not already owned.
-	for _, machine := range machines {
-		if shouldAdopt(machineSet, machine) {
-			c.adoptOrphan(machineSet, machine)
-		}
-	}
-
-	var result error
-	currentMachineCount := int32(len(machines))
-	desiredReplicas := *machineSet.Spec.Replicas
-	diff := int(currentMachineCount - desiredReplicas)
-
+func (c *MachineSetControllerImpl) syncReplicas(ms *v1alpha1.MachineSet, machines []*v1alpha1.Machine) error {
+	diff := len(machines) - int(*(ms.Spec.Replicas))
 	if diff < 0 {
 		diff *= -1
+		glog.V(4).Infof("Too few replicas for %v %s/%s, need %d, creating %d", controllerKind, ms.Namespace, ms.Name, *(ms.Spec.Replicas), diff)
+
+		var errstrings []string
 		for i := 0; i < diff; i++ {
-			glog.V(2).Infof("creating a machine ( spec.replicas(%d) > currentMachineCount(%d) )", desiredReplicas, currentMachineCount)
-			machine := c.createMachine(machineSet)
-			_, err := c.clusterAPIClient.ClusterV1alpha1().Machines(machineSet.Namespace).Create(machine)
+			glog.V(2).Infof("creating a machine ( spec.replicas(%d) > currentMachineCount(%d) )", *(ms.Spec.Replicas), len(machines))
+			machine := c.createMachine(ms)
+			_, err := c.clusterAPIClient.ClusterV1alpha1().Machines(ms.Namespace).Create(machine)
 			if err != nil {
 				glog.Errorf("unable to create a machine = %s, due to %v", machine.Name, err)
-				result = err
+				errstrings = append(errstrings, err.Error())
 			}
 		}
+
+		if errstrings != nil {
```

Comment:
(sanity check)
What is the default empty value for []string? is it nil or is it an empty list?

Reply:
If empty list, this condition might not work as you expect.

Reply:
the default value of an empty slice is equal to nil

Reply:
changed it to make it easier to read